static/src/javascripts/projects/common/modules/article/spacefinder.js

```diff
@@ -57,46 +62,38 @@ define([
         // fromBotton:Boolean
         // will reverse the order of slots (this is useful for lazy loaded content)
         fromBottom: false
-    },
-    imagesLoaded,
-    richLinksUpgraded;
+    };
+
+    function expire(resolve) {
+        window.setTimeout(resolve, LOADING_TIMEOUT);
+    }
 
     function onImagesLoaded(body) {
         var notLoaded = filter(qwery('img', body), function (img) {
             return !img.complete;
         });
 
-        return imagesLoaded || (imagesLoaded = Promise.race([
-            Promise.all(map(notLoaded, function (img) {
-                return new Promise(function (resolve) {
-                    bean.on(img, 'load', resolve);
-                });
-            })),
+        return notLoaded.length === 0 ?
+            Promise.resolve(true) :
             new Promise(function (resolve) {
-                window.setTimeout(resolve, 5000);
-            })
-        ]));
+                var loadedCount = 0;
+                bean.on(body, 'load', notLoaded, function onImgLoaded() {
+                    loadedCount += 1;
+                    if (loadedCount === notLoaded.length) {
+                        bean.off(body, 'load', onImgLoaded);
+                        notLoaded = null;
+                        resolve();
+                    }
+                });
+            });
     }
 
     function onRichLinksUpgraded(body) {
-        return richLinksUpgraded || (richLinksUpgraded = Promise.race([
-            new Promise(function (resolve, reject) {
-                var unloaded = qwery('.element-rich-link--not-upgraded', body);
-
-                if (!unloaded.length) {
-                    resolve();
-                } else {
-                    reject();
-                }
-            }).catch(function () {
-                return new Promise(function (resolve) {
-                    mediator.once('rich-link:loaded', resolve);
-                });
-            }),
+        return qwery('.element-rich-link--not-upgraded', body).length === 0 ?
```

Comment:
Is this class used for styling too? We like to use js-hook but I can imagine that in this situation it maybe weird to do?!